src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -911,9 +917,10 @@ static PyObject *TopicPartition_str0 (TopicPartition *self) {
 
 	ret = cfl_PyUnistr(
 		_FromFormat("TopicPartition{topic=%s,partition=%"CFL_PRId32
-			    ",offset=%s,error=%s}",
+			    ",offset=%s,metadata=%s,error=%s}",
```

Comment:
The TopicPartition is used for different purposes, like assign() where the metadata is not relevant, so the metadata is not always relevant.
Secondly, the metadata is binary so it will not be printable. Suggest not including it in the stringification at all.